src/meta/test/AdminClientTest.cpp

```diff
@@ -0,0 +1,263 @@
+    }
+
+    folly::Future<storage::cpp2::AdminExecResp>
+    future_addPart(const storage::cpp2::AddPartReq& req) override {
+        RETURN_OK(req);
+    }
+
+    folly::Future<storage::cpp2::AdminExecResp>
+    future_addLearner(const storage::cpp2::AddLearnerReq& req) override {
+        RETURN_OK(req);
+    }
+
+    folly::Future<storage::cpp2::AdminExecResp>
+    future_waitingForCatchUpData(const storage::cpp2::CatchUpDataReq& req) override {
+        RETURN_OK(req);
+    }
+
+    folly::Future<storage::cpp2::AdminExecResp>
+    future_removePart(const storage::cpp2::RemovePartReq& req) override {
+        RETURN_OK(req);
+    }
+
+    folly::Future<storage::cpp2::AdminExecResp>
+    future_memberChange(const storage::cpp2::MemberChangeReq& req) override {
+        RETURN_OK(req);
+    }
+};
+
+class TestStorageServiceRetry : public TestStorageService {
+public:
+    TestStorageServiceRetry(IPv4 ip, Port port) {
+        leader_.set_ip(ip);
+        leader_.set_port(port);
+    }
+
+    folly::Future<storage::cpp2::AdminExecResp>
+    future_addLearner(const storage::cpp2::AddLearnerReq& req) override {
+        RETURN_LEADER_CHANGED(req, leader_);
+    }
+
+    folly::Future<storage::cpp2::AdminExecResp>
+    future_waitingForCatchUpData(const storage::cpp2::CatchUpDataReq& req) override {
+        RETURN_LEADER_CHANGED(req, leader_);
+    }
+
+    folly::Future<storage::cpp2::AdminExecResp>
+    future_memberChange(const storage::cpp2::MemberChangeReq& req) override {
+        RETURN_LEADER_CHANGED(req, leader_);
+    }
+
+private:
+    nebula::cpp2::HostAddr leader_;
+};
+
+TEST(AdminClientTest, SimpleTest) {
+    auto sc = std::make_unique<test::ServerContext>();
+    auto handler = std::make_shared<TestStorageService>();
+    sc->mockCommon("storage", 0, handler);
+    LOG(INFO) << "Start storage server on " << sc->port_;
+
+    uint32_t localIp;
+    network::NetworkUtils::ipv4ToInt("127.0.0.1", localIp);
+    fs::TempDir rootPath("/tmp/AdminTest.XXXXXX");
+    std::unique_ptr<kvstore::KVStore> kv(TestUtils::initKV(rootPath.path()));
+    auto client = std::make_unique<AdminClient>(kv.get());
+
+    {
+        LOG(INFO) << "Test transLeader...";
+        folly::Baton<true, std::atomic> baton;
+        client->transLeader(0, 0, {localIp, sc->port_}).then([&baton](auto&& st) {
+            CHECK(st.ok());
+            baton.post();
+        });
+        baton.wait();
+    }
+    {
+        LOG(INFO) << "Test addPart...";
+        folly::Baton<true, std::atomic> baton;
+        client->addPart(0, 0, {localIp, sc->port_}, true).then([&baton](auto&& st) {
+            CHECK(st.ok());
+            baton.post();
+        });
+        baton.wait();
+    }
+    {
+        LOG(INFO) << "Test removePart...";
+        folly::Baton<true, std::atomic> baton;
+        client->removePart(0, 0, {localIp, sc->port_}).then([&baton](auto&& st) {
+            CHECK(st.ok());
+            baton.post();
+        });
+        baton.wait();
+    }
```

Comment:
what about add part again?
or let's loop the test a few times

Reply:
AddPart again will be tested on server side.  The UT is just to test client interface.

Reply:
Yep. It's simple for the client,  but a valued corned case.